arwen/contexts/validator.go

```diff
@@ -34,36 +36,48 @@ func (validator *WASMValidator) verifyFunctions(instance *wasmer.Instance) error
 			return fmt.Errorf("%w: %s", arwen.ErrInvalidFunctionName, functionName)
 		}
 
-		if !validator.isVoidFunction(instance, functionName) {
+		isVoid, err := validator.isVoidFunction(instance, functionName)
+		if err == arwen.ErrFuncNotFound {
+			return fmt.Errorf("%w: %s", arwen.ErrFuncNotFound, functionName)
+		}
```

Comment:
Else, other kind of error? I see that it cannot happen, therefore the condition is true anyway if err is not nil? Perhaps define and return another error, such as `ErrFuncVerificationFailed`?